core/src/main/java/org/apache/iceberg/deletes/Deletes.java

```diff
@@ -158,7 +164,7 @@ protected boolean shouldKeep(T row) {
     }
   }
 
-  private static class PositionStreamDeleteFilter<T> extends CloseableGroup implements CloseableIterable<T> {
+  protected static class PositionStreamDeleteFilter<T> extends CloseableGroup implements CloseableIterable<T> {
```

Comment:
I think this can still be private because the subclass is also defined in this file.